core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -50,17 +65,68 @@ public FilteredQueryBuilder(@Nullable QueryBuilder queryBuilder, @Nullable Query
         this.filterBuilder = filterBuilder;
     }
 
+    /** Returns the query to apply the filter to. */
+    public QueryBuilder query() {
+        return queryBuilder;
+    }
+
+    /** Returns the filter to apply to the query results. */
+    public QueryBuilder filter() {
+        return filterBuilder;
+    }
+
     @Override
-    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject(NAME);
-        if (queryBuilder != null) {
-            builder.field("query");
-            queryBuilder.toXContent(builder, params);
+    protected boolean doEquals(FilteredQueryBuilder other) {
+        return Objects.equals(queryBuilder, other.queryBuilder) &&
+                Objects.equals(filterBuilder, other.filterBuilder);
+    }
+
+    @Override
+    public int doHashCode() {
+        return Objects.hash(queryBuilder, filterBuilder);
+    }
+
+    @Override
+    public Query doToQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query result = null;
+
+        Query query = queryBuilder.toQuery(parseContext);
+        Query filter = filterBuilder.toQuery(parseContext);
+
+        if (query == null) {
+            return null;
```

Comment:
Also think this is right here. In JSON you get null here for
```
{
  "query": {
    "filtered": { 
      "query": { },
      "filter": {
        "range": { "created": { "gte": "now - 1d / d" }}
      }
    }
  }
}
```
 Which means the user specified something as query but its the empty set. In this case its not clear what to filter, but also not save to return any of Match_All/No_Docs query, since that would mean something different depending on any (potential) enclosing query, e.g. when this is used in "must" or in "must_not".